client/utils/src/mpsc.rs

```diff
@@ -76,103 +68,78 @@ mod inner {
-				queue_size: self.queue_size.clone(),
 				queue_size_warning: self.queue_size_warning,
 				warning_fired: self.warning_fired.clone(),
 				creation_backtrace: self.creation_backtrace.clone(),
 			}
 		}
 	}
 
-	/// Wrapper Type around `UnboundedReceiver` that decreases the global
+	/// Wrapper Type around `async_channel::Receiver` that decreases the global
 	/// measure when a message is polled
 	#[derive(Debug)]
 	pub struct TracingUnboundedReceiver<T> {
-		inner: UnboundedReceiver<T>,
+		inner: Receiver<T>,
 		name: &'static str,
-		queue_size: Arc<AtomicI64>,
 	}
 
-	/// Wrapper around `mpsc::unbounded` that tracks the in- and outflow via
+	/// Wrapper around `async_channel::unbounded` that tracks the in- and outflow via
 	/// `UNBOUNDED_CHANNELS_COUNTER` and warns if the message queue grows
 	/// above the warning threshold.
 	pub fn tracing_unbounded<T>(
 		name: &'static str,
-		queue_size_warning: i64,
+		queue_size_warning: usize,
 	) -> (TracingUnboundedSender<T>, TracingUnboundedReceiver<T>) {
-		let (s, r) = mpsc::unbounded();
-		let queue_size = Arc::new(AtomicI64::new(0));
+		let (s, r) = async_channel::unbounded();
 		let sender = TracingUnboundedSender {
 			inner: s,
 			name,
-			queue_size: queue_size.clone(),
 			queue_size_warning,
 			warning_fired: Arc::new(AtomicBool::new(false)),
-			creation_backtrace: Arc::new(Backtrace::new_unresolved()),
+			creation_backtrace: Arc::new(Backtrace::force_capture()),
 		};
-		let receiver = TracingUnboundedReceiver { inner: r, name, queue_size };
+		let receiver = TracingUnboundedReceiver { inner: r, name };
 		(sender, receiver)
 	}
 
 	impl<T> TracingUnboundedSender<T> {
-		/// Proxy function to mpsc::UnboundedSender
-		pub fn poll_ready(&self, ctx: &mut Context) -> Poll<Result<(), SendError>> {
-			self.inner.poll_ready(ctx)
-		}
-
-		/// Proxy function to mpsc::UnboundedSender
+		/// Proxy function to `async_channel::Sender`
 		pub fn is_closed(&self) -> bool {
 			self.inner.is_closed()
 		}
 
-		/// Proxy function to mpsc::UnboundedSender
-		pub fn close_channel(&self) {
-			self.inner.close_channel()
-		}
-
-		/// Proxy function to mpsc::UnboundedSender
-		pub fn disconnect(&mut self) {
-			self.inner.disconnect()
-		}
-
-		pub fn start_send(&mut self, msg: T) -> Result<(), SendError> {
-			// The underlying implementation of [`UnboundedSender::start_send`] is the same as
-			// [`UnboundedSender::unbounded_send`], so we just reuse the message counting and
-			// error reporting code from `unbounded_send`.
-			self.unbounded_send(msg).map_err(TrySendError::into_send_error)
+		/// Proxy function to `async_channel::Sender`
+		pub fn close(&self) -> bool {
+			self.inner.close()
 		}
 
-		/// Proxy function to mpsc::UnboundedSender
-		pub fn unbounded_send(&self, msg: T) -> Result<(), TrySendError<T>> {
-			self.inner.unbounded_send(msg).map(|s| {
+		/// Proxy function to `async_channel::Sender`
+		pub fn try_send(&self, msg: T) -> Result<(), TrySendError<T>> {
```

Comment:
Can we not stay with `unbounded_send`? 

I mean we don't need to try sending, we are having here an unbounded channel that only rejects when the receiver is dropped. So, `try` sounds wrong to me here

Reply:
> Can we not stay with `unbounded_send`?

Done